src/Utils/TypeInfo.php

```diff
@@ -340,11 +342,12 @@ static function ($arg) use ($node) {
                 }
                 $this->argument            = $argDef;
                 $this->defaultValueStack[] = $argDef && $argDef->defaultValueExists() ? $argDef->defaultValue : Utils::undefined();
-                $this->inputTypeStack[]    = Type::isInputType($argType) ? $argType : null;
+                $this->inputTypeStack[]    = $argType;
                 break;
 
             case $node instanceof ListValueNode:
-                $listType = Type::getNullableType($this->getInputType());
+                $type     = $this->getInputType();
+                $listType = $type !== null ? Type::getNullableType($type) : null;
```

Comment:
cognitive load can be decreased

```suggestion
                $listType = $type === null ? null : Type::getNullableType($type);
```